api/proto/services/service.proto

```diff
@@ -0,0 +1,28 @@
+syntax = "proto3";
+
+import "objects/state.proto";
+import "rpc/read_revisions.proto";
+import "rpc/snapshot.proto";
+
+package jj_api.services;
+
+// All jj things will go into this service.
+service JjService {
+  // This is generally the first rpc you should be calling.
+  // It deviates from the norm by returning an object rather than a response so
+  // that you can pass that in to options in later requests.
+  rpc Snapshot(jj_api.rpc.SnapshotRequest) returns (jj_api.objects.RepoState) {}
```

Comment:
I went back and forth on this, but I think you raise a good point. In general, I believe that while a user interface should have reasonable implicit defaults, APIs should be explicit. I think there's a decent middle-ground though, where we allow a `RevisionsRequest` to do it all in one RPC, but it will throw an error if they don't explicitly say whether they want to snapshot or not.

I do like your `Begin/EndTransaction` idea for read-write operations (which I haven't really considered yet), so I've fleshed out how I think that might look.

My design is, roughly:
* You can create a transaction from a `RepoState`.
* Write operations will require a transaction ID to be set
* Readonly operations will require *either* a transaction ID or a `RepoState`, since they don't require a transaction to have begun.